src/libraries/System.Private.CoreLib/src/System/IO/StringReader.cs

```diff
@@ -46,9 +43,9 @@ public override int Peek()
         {
             if (_s == null)
             {
-                throw new ObjectDisposedException(null, SR.ObjectDisposed_ReaderClosed);
+                ThrowHelper.ThrowObjectDisposedException_ReaderClosed();
             }
-            if (_pos == _length)
+            if (_pos == _s.Length)
```

Comment:
As long as we're changing the code to streamline it, presumably it'd be even more so as something like:
```C#
public override int Peek()
{
    string? s = _s;
    if (s == null)
    {
        ThrowHelper.ThrowObjectDisposedException_ReaderClosed();
    }

    int pos = _pos;
    if ((uint)pos < s.Length)
    {
        return s[pos];
    }

    return -1;
}
```
?